### ISSUE TITLE
[Feature Request]: Even better support on Volt

### ISSUE THREAD

**Issue by nunomaduro** [laravel-idea/plugin#819]
### Feature Description

First and foremost, congratulations on making Volt almost fully compatible with PHPStorm. However, I've detected a few missing pieces:

```php
<?php

use App\Livewire\Forms\PostForm;

use function Livewire\Volt\{action, form, state, with};


state('counter'); // ❌ missing $counter on the view. note: counter is "null"...
state(['counter' => 0]); // ✅
state('counter', 0); // ❌ missing $counter on the view. 
state(counter: 0, /** or more */); // ❌ missing $counter on the view. 

form(PostForm::class); // ❌ missing $this->form withing callbacks like $increment...
form(PostForm::class, 'my_form'); // ❌ missing $this->my_form withing callbacks like $increment...

uses(MyTrait::class); // ❌ check point 3 below...

$increment = function () {
    $this->     // ❌ should give access to methods from:
                // - 1: \Livewire\Volt\Component (and extended classes)...
                // - 2: \Illuminate\Foundation\Auth\Access trait...
                // - 3: Any trait given to `uses` function...
});

/** 1. closure */
with(fn () => [
    'posts' => $this->posts->all(),  // ❌ missing $posts on the view
    //
]);

with(fn () => [
    'posts' => 1, // ❌ missing $posts on the view
]);

with(posts: fn () => 1, /** ... */); // ❌ missing $posts on the view
with(['posts' => fn () => 1, /** ... */]); // ❌ missing $posts on the view
with('posts', fn () => 1); // ❌ missing $posts on the view
```

**Comment by adelf** [comment]
Thanks) I can copy this to my tests and just make them green!

**Comment by robsontenorio** [comment]
@adelf I am using Volt class based API. 

It would possible to **avoid** auto import FQDN classes on `<?php ... ?>` block?

see #838

**Comment by adelf** [comment]
Implemented in 7.4. Thank you for the ideal description)
